### PR TITLE
feat: Add styling params to common_TitleBar

### PR DESCRIPTION
**Pull request by jbrundage**
Add publish proxy params for the ChartPanel TitleBar

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage - is this related to:  https://github.com/hpcc-systems/Visualization/issues/3176?

File: packages/common/src/TitleBar.ts [hunk 1]
```diff
@@ -54,16 +64,55 @@ export class Button extends HTMLWidget {
...
+            };
+        }
+        return {
+            x: (round ? Math.round(this._boundingBox.x) : this._boundingBox.x) * this._widgetScale,
+            y: (round ? Math.round(this._boundingBox.y) : this._boundingBox.y) * this._widgetScale,
+            width: (round ? Math.round(this._boundingBox.width) : this._boundingBox.width) * this._widgetScale,
+            height: (round ? Math.round(this._boundingBox.height) : this._boundingBox.height) * this._widgetScale
+        };
+    }
+
```
Comment (GordonSmith) [inline packages/common/src/TitleBar.ts]: How is this different to the default version in HTML Widget?

File: packages/common/src/TitleBar.ts [hunk 2]
```diff
@@ -34,6 +36,14 @@ export class Button extends HTMLWidget {
         element
             .classed("disabled", !this.enabled())
             .attr("title", this.tooltip())
+            .style("display", "inline-block")
+            .style("height", this.fixedHeight() + "px")
+            .style("line-height", this.fixedHeight() + "px")
+            .style("min-width", this.fixedHeight() + "px")
```
Comment (GordonSmith) [inline packages/common/src/TitleBar.ts]: Is this deliberate?

Comment (jbrundage) [reply packages/common/src/TitleBar.ts]: Yes, it ensures the buttons are square (or wider if the text content needs it to be)

File: packages/common/src/TitleBar.ts (hunk 1 above)
Comment (jbrundage) [reply packages/common/src/TitleBar.ts]: The default version in HTML Widget returns the bbox of `this._element.node().firstElementChild`. That was causing problems with TitleBar.

Comment (GordonSmith) [reply packages/common/src/TitleBar.ts]: I saw that, but was wondering why?  (probably related to title bar not wanting to 100% high like most widgets?)

That being said, you could add a new method to the HTMLWIdget:
```
calcBBox(domNode: HTMLElement, refresh:boolean = false, round:boolean = false) {
...
}
```
And have both your variants call it?

File: packages/common/src/TitleBar.ts (hunk 2 above)
Comment (GordonSmith) [reply packages/common/src/TitleBar.ts]: Ok (just checking)

**Comment by jbrundage** [comment]
> @jbrundage - is this related to: #3176?

It's not related to that issue, but I'll work on that one after the CL IE issue that just came in

File: packages/common/src/TitleBar.ts (hunk 1 above)
Comment (jbrundage) [reply packages/common/src/TitleBar.ts]: Sounds good

File: packages/common/src/TitleBar.ts [hunk 3]
```diff
@@ -8,6 +8,8 @@ import "../src/TitleBar.css";
 export class Button extends HTMLWidget {
     private _enabled = true;
 
+    _iconElement;
+
```
Comment (GordonSmith) [inline packages/common/src/TitleBar.ts]: Should be private or public and have a type

File: packages/common/src/TitleBar.ts [hunk 4]
```diff
@@ -148,6 +191,18 @@ export class IconBar extends HTMLWidget {
...
+            }
+            if ((item as any).fontSizeStyle) {
+                (item as any).fontSizeStyle(this.buttonFontSizeStyle());
+            }
+            if ((item as any).lineHeightStyle) {
+                (item as any).lineHeightStyle(this.buttonLineHeightStyle());
+            }
+            if ((item as any).margin) {
+                (item as any).margin(this.buttonMargin());
+            }
```
Comment (GordonSmith) [inline packages/common/src/TitleBar.ts]: These should be inside a: 
```
else if (item instanceof Button) {
...
}
```
Then you won't need to test for each one individually...

**Comment by GordonSmith** [comment]
I think I would like to take a look at the issue that this is solving.  I am not sure the title bar should be mandating some of the button sizes the way it is.

**Pull request closed without merging**